text/orderer-v3.md

```diff
@@ -0,0 +1,320 @@
+
+The `signature` is the signature that is verifiable under the initiator's public key over the ASN1 encoding of `<version, timestamp, from_id, to_id, session_id, channel>`.
+
+The `version` field determines what fields the signature was computed over, and the current version is zero. 
+
+The `session_binding` prevents MITM (Man In The Middle) attacks, and it is explained next.
+
+
+### Protecting against Man-In-The-Middle attacks
+
+The safety of the protocol described in the previous paragraph cannot be guaranteed in case the transport layer is hijacked by an adversary. 
+Consider a scenario where an adversary obtains the private key that corresponds to the TLS certificate of the responder node. 
+The adversary can hijack the network traffic between the initiator and the responder, wait for the `Auth` message, and then impersonate the 
+initiator before the responder and impersonate the responder before the initiator by relaying the `Auth` through itself.
+
+As noted earlier, TLS certificate private keys are more susceptible for theft as they might not be stored in HSM.
+In such a case, it is still possible to detect a MITM attack by taking advantage of [Keying Material Exporters](https://datatracker.ietf.org/doc/html/rfc5705).
+The [ExportKeyingMaterial](https://pkg.go.dev/crypto/tls#ConnectionState.ExportKeyingMaterial) API allows both ends of a TLS connection to derive a shared key from the
+`client_random`, `server_random`, `master_secret` used to establish the TLS session. Since this key is identical to the initiator and responder, 
+it can be used by the responder to detect whether the message is being relayed by an attacker. 
+
+The detection is quite simple: The initiator node and the receiver node both compute: `r = Hash(version || timestamp || from_id || to_id, channel)`
+and then `session_binding = ExportKeyingMaterial(r)`. The initiator node encodes `session_binding` in the `Auth` message, and the responder node 
+expects the `session_binding` from the `Auth` message to match what it computed on its own.
+
+In case an attacker manages to obtain the TLS private key of the responder node and performs a MITM attack, the function `ExportKeyingMaterial`
+on the initiator node and the real responder node would yield different values, because the `master_secret` will differ.
+The `master_secret` will differ because Fabric orderers will be restricted to use ECDHE key exchange. With ECHDE key exchange, a MITM with the same `master_secret` 
+across both the initiator and responder means that the initiator and responder each picked the public key share, but the attacker managed to compute the shared key,
+thus breaking the [CDH assumption](https://en.wikipedia.org/wiki/Computational_Diffie%E2%80%93Hellman_assumption).
+
+In case the responder node is located behind a TLS terminating reverse proxy, it can be configured to ignore the `session_binding` check, 
+but the check will be enabled by default.
+
+# Testing
```

Comment:
For a change of this scale, I think this section should include "Implementation Plan and Testing".

For example will the existing fabric repo be updated to support both the old v2.x orderer node and new v3.0 orderer node in parallel?

Will the existing fabric repo support Solo and Raft and SmartBFT orderer binaries? Or will each of the orderer implementations be delivered in different repositories that import the base Fabric orderer Go modules?

Will the TLS pinning approach continue to be supported to enable rolling upgrades, perhaps in a deprecated mode so that it can be retired in a future release?

Reply:
Addressed the comments modulo BFT which is not discussed in that RFC.